### ISSUE TITLE
Feature: Package size

### ISSUE THREAD

**Issue by vanlooverenkoen** [dart-lang/pub-dev#8114]
It would be nice to know what the actual size is of a package. And see this on pub.dev.

2 type of sizes would be interesting
- Size excl dependencies
- Size incl dependencies

**Comment by sigurdm** [comment]
I agree it would be nice to list the archive sizes on the "versions" page.

However remember that (given dart's treeshaking) the size affect on the actual produced app is largely unrelated to the size of the package.

> Size incl dependencies

This is a lot harder to make useful - dependencies can often be shared. What would be the use case for this?

**Comment by vanlooverenkoen** [comment]
Fair point. But at least you have an idea of the size of the package without actually digging into the code. Just an idea

**Comment by sigurdm** [comment]
Maybe just listing the number of transitive dependencies of a package (in the default resolution) somewhere would be useful...

**Comment by isoos** [comment]
> Maybe just listing the number of transitive dependencies of a package (in the default resolution) somewhere would be useful...

We have that in the scoring page, after the direct dependencies:
<img width="613" alt="image" src="https://github.com/user-attachments/assets/aa9ef558-f6c1-4d34-8ed4-aaac9d7e19fc">

However, it changes over time (as the dependent packages may change) and it also depends on the SDK being used for the analysis.

**Comment by isoos** [comment]
Also: what size are we interested in? gzipped archive size, uncompressed total size (incl/excluding tests and other assets)? It can vary based on the needs and uses of a package.
I think we should expose the single archive file size, and maybe the total archive file size, but other than that, the analysis gets into a grey area.

**Comment by sigurdm** [comment]
For the version listing I would be interested in the compressed archive size.

Maybe the others are useful for some purposes. But yeah - a very grey area.
